controllers/volumereplicationgroup_controller.go

```diff
@@ -112,24 +111,63 @@ func pvcPredicateFunc() predicate.Funcs {
 
 			log.Info("Update event for PersistentVolumeClaim")
 
-			// TODO: If finalizers change then deep equal of spec fails to catch it, we may want more
-			// conditions here, compare finalizers and also status.state to catch bound PVCs
-			return !reflect.DeepEqual(oldPVC.Spec, newPVC.Spec)
+			return updateEventDecision(oldPVC, newPVC, log)
 		},
 		DeleteFunc: func(e event.DeleteEvent) bool {
-			log := ctrl.Log.WithName("pvcmap").WithName("VolumeReplicationGroup")
-
-			log.Info("delete event for PersistentVolumeClaim")
-
-			// PVC deletes are not of interest, as the added finalizers by VRG would be removed
-			// when VRGs are deleted, which would trigger the delete of the PVC.
+			// No need to reconcile VRG on pvc deletion.
```

Comment:
Update comment to state something like so?:
PVC deletion is held back till VRG deletion, to avoid races between, PVC deletion during Subscription cleanup and VRG state updates. For example, if VRG state is not updated prior to Subscription cleanup a PVC deletion would result in leaving the VR with a stale state as per the existing VRG state.

Reply:
ACK